src/impl/vamp/vector/interface.hh

```diff
@@ -459,8 +459,11 @@ namespace vamp
         template <typename OtherT, typename BoundsT>
         inline static constexpr auto map_to_range(OtherT v, BoundsT min_v, BoundsT max_v) noexcept -> D
         {
+            constexpr float lo = -0.5F;
+            constexpr float hi = 0.5F;
             const auto normalized = D(apply<S::template map_to_range<typename OtherT::S::VectorT>>(v.data));
-            return min_v + (normalized * (max_v - min_v));
+
+            return min_v + ((normalized - lo) / (hi - lo)) * (max_v - min_v);
         }
```

Comment:
Was this broken before? We should also be using the scalar types specified by the vector interface rather than hardcoding `float`.